src/software/ai/hl/stp/tactic/get_possession/get_possession_fsm.h

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include "shared/constants.h"
+#include "software/ai/evaluation/pass.h"
+#include "software/ai/hl/stp/tactic/move/move_fsm.h"
+#include "software/ai/hl/stp/tactic/tactic.h"
+#include "software/ai/intent/move_intent.h"
+#include "software/geom/algorithms/acute_angle.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/geom/algorithms/distance.h"
+
+struct GetPossessionFSM
+{
+    class InterceptBallState;
+
+    struct ControlParams
+    {
+    };
+
+    DEFINE_UPDATE_STRUCT_WITH_CONTROL_AND_COMMON_PARAMS
+
+    static constexpr double BALL_MOVING_SLOW_SPEED_THRESHOLD = 0.3;
+
+    /**
+     * Calculates the interception point for intercepting balls
+     *
+     * @param robot The robot to do the interception
+     * @param ball The ball to intercept
+     * @field The field to intercept on
+     *
+     * @return the best interception point
+     */
+    static Point findInterceptionPoint(const Robot &robot, const Ball &ball,
+                                       const Field &field)
```

Comment:
can this live in the evaluation folder?

Reply:
https://github.com/UBC-Thunderbots/Software/issues/1968